earth_enterprise/src/fusion/gepublish/PublisherClient.cpp

```diff
@@ -1221,40 +1269,38 @@ bool PublisherClient::UploadFiles(ServerType server_type,
 
   int64 processed_size = 0;
   if (is_server_host_same_as_publishing)  {
+    notify(NFY_DEBUG, "Transfering files locally");
     size_t num_entries = entries.size();
-    for (size_t i = 0; i < num_entries; ++i) {
+    std::vector<ManifestEntry> dependent_entries;
+    size_t i = 0;
+    for (; i < num_entries; ++i) {
       const ManifestEntry& entry = entries[i];
-      std::string src_path(entry.current_path);
-      // Note: all files coming to upload should exist.
-      assert(khExists(src_path));
-      std::string dest_path(server_prefix + "/" +
-                            host_root + entry.orig_path);
-      assert(src_path != dest_path);  // We send only delta always
-      // The logic for prefer_copy was fixed in 3.0.3.
-      // It used to set prefer_copy = !tmp_dir.empty(), when
-      // in fact the desired effect is only to prefer copies for
-      // source files that are in the tmp_dir, and to defer to the
-      // server setting for AllowSymLinks: Y in PUBLISH_ROOT/.config.
-
-      // If the source file is in the temp directory, we want to copy (hard
-      // link is OK whenever possible).
-      // Otherwise, a symbolic link is OK whenever possible.
-      bool prefer_copy = !tmpdir.empty() && src_path.find(tmpdir) == 0;
-
-      // Many LocalTransfers in succession may sometimes overwhelm a server.
-      // To be safe we do a handful of retries with increasing delays between
-      // attempts.
-      int tries = 4;
-      int sleep_secs = 15;
-      while (!LocalTransfer(
-                 server_type, src_path, dest_path, prefer_copy)) {
-        if (--tries == 0) {
+
+      if (!LocalTransferWithRetry(server_prefix, host_root, server_type,
+                                  tmpdir, entry.current_path, entry.orig_path))
+        return false;
+
+      for(size_t iDep = 0; iDep < entry.dependents.size(); ++iDep) {
+        notify(NFY_DEBUG, "Adding dependent file: %s", entry.dependents[iDep].c_str());
+        dependent_entries.push_back(ManifestEntry(entry.dependents[iDep]));
+      }
+
+      if (report_progress && progress_ != NULL) {
```

Comment:
I think there might be an error in the progress reporting. The first time I pushed it went to about 99%, then dropped back to about 50%, then went to 100%. It happened really quickly so I'm not completely sure of what I saw, but there may be some error in how this reports progress now.

Reply:
I'll look into it.  There might be a pre-loop somewhere calculating total size.  I would need to fix it to add in the size of the data files.

Reply:
This was actually a good find...  I think my code was double transferring files when for local server.  Ugh...  I'll clean that up.